### PR TITLE
Add support for extra AWS resource filters on the CLI

### PR DESCRIPTION
**Pull request by a1exsh**
This helps to limit update or remote command execution operations to subsets
of the nodes in a cluster, such as Availability Zones, etc.

### CODE REVIEW

**Comment by a1exsh** [comment]
:+1:

File: planb/cli.py [hunk 1]
```diff
@@ -35,12 +36,17 @@ def validate_dc_suffix(ctx, param, value):
     return value
 
 
+def parse_filters(ctx, param, value):
+    return json.loads(value)
```
Comment (slitsche) [inline planb/cli.py]: Why this long list of parameters if we use only value?

Comment (slitsche) [reply planb/cli.py]: And could we add a validation so that it is ensured the result is a list of objects? (couldn't see this validation elsewhere)

**Comment by a1exsh** [comment]
:+1:

File: planb/cli.py [hunk 2]
```diff
@@ -35,12 +36,32 @@ def validate_dc_suffix(ctx, param, value):
...
+
+def validate_filters(ctx, param, value):
+    try:
+        filters = json.loads(value)
+    except Exception as e:
+        raise click.BadParameter("JSON parsing error: {}".format(e))
+
+    if not (isinstance(filters, list) and
+            all(is_valid_filter(f) for f in filters)):
+        raise click.BadParameter("has to match schema {}".format(filters_schema_doc))
```
Comment (slitsche) [inline planb/cli.py]: Start with uppercase ? Maybe s/has/Filter has/

Comment (a1exsh) [reply planb/cli.py]: This is how the error message looks like: `Error: Invalid value for "--filters": has to match schema [{"Name":"string","Values":["string",...]},...]`

See also existing validations: https://github.com/zalando-stups/planb-cassandra/pull/239/files/a60962232bf808b79e85645f8ac79364b83af90c#diff-009a217c30d606342b4db9f6782bca0cL27

**Comment by slitsche** [comment]
:+1:

**Pull request merged**